docs/src/main/tut/chisel3/interfaces-and-connections.md

```diff
@@ -129,3 +132,14 @@ class ConsumingData extends Module {
   // do something with io.readyValid.bits
 }
 ```
+
+Decoupled is marker interface for a ready-valid interface to indicate
+that there are no requirements placed on the signaling of `ready` or `valid`.
+That means `ready` and `valid` can also be de-asserted without a data transfer.
+
+Irrevocable is marker interface for a ready-valid interface to indicate
+that the value of 'bits' after a cycle where 'valid' is high and 'ready' is low
+are not changed. E.g., `valid` stays asserted. Also the consumer shall
+keep `ready` asserted after a cycle where `read` was high and `valid` was low.
```

Comment:
I think this is wrong. The convention is that `valid` will stay high and not de-assert. I don't think there's any guarantees on `ready`?